packages/flame_3d/lib/src/resources/light/light.dart

```diff
@@ -0,0 +1,24 @@
+import 'package:flame_3d/game.dart';
+import 'package:flame_3d/graphics.dart';
+import 'package:flame_3d/resources.dart';
+
+/// {@template light}
+/// A [Resource] that represents a light source that changes how the scene is
+/// rendered.
+///
+/// This class isn't a true resource, it does not upload it self to the GPU.
+/// Instead, it is used to modify how meshes are uploaded.
+/// {@endtemplate}
+class Light extends Resource<void> {
+  // TODO(luan): Add more attributes and light types
```

Comment:
In the long run we can make `Resource` a macro and allowing us to have more dynamic structs for shaders.